src/app/public/modules/repeater/repeater-adapter.service.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ElementRef,
+  Injectable
+} from '@angular/core';
+
+import {
+  SkyRepeaterService
+} from './repeater.service';
+
+@Injectable()
+export class SkyRepeaterAdapterService {
+  private host: ElementRef;
+
+  constructor(private repeaterService: SkyRepeaterService) {}
+
+  public setRepeaterHost(hostRef: ElementRef): void {
+    this.host = hostRef;
+  }
+
+  public moveItemUp(elementRef: ElementRef, top = false, steps = 1): number {
+    const itemArray = Array.from(this.host.nativeElement.querySelectorAll('sky-repeater-item'));
+    const index = itemArray.indexOf(elementRef.nativeElement);
+
+    if (index === 0) {
+      return;
+    }
+
+    let newIndex = index - steps;
+
+    if (top || newIndex < 0) {
+      newIndex = 0;
+    }
+
+    return this.moveItem(elementRef, index, newIndex);
+  }
+
+  public moveItemDown(elementRef: ElementRef, steps = 1): number {
+    const itemArray = Array.from(this.host.nativeElement.querySelectorAll('sky-repeater-item'));
+    const index = itemArray.indexOf(elementRef.nativeElement);
```

Comment:
Not a deal breaker, but would it be nice to encaspulate the query selector getters so we don't repeater ouselves? For example:
```
private getItems()
private getRepeater()
```

Reply:
So I see the value in `getItems` but I don't really see it on the repeater. We only make that call once.  We can extract it down the road if we end up doing the same in future methods.